networks/suzuka/suzuka-full-node/src/partial.rs

```diff
@@ -232,23 +239,71 @@ where
 	// ! Currently this only implements opt.
 	/// Runs the executor until crash or shutdown.
 	async fn run_executor(&self) -> Result<(), anyhow::Error> {
+		debug!("SuzukaPartialNode run_executor");
 		// wait for both tasks to finish
 		tokio::try_join!(self.write_transactions_to_da(), self.read_blocks_from_da())?;
 
 		Ok(())
 	}
+
+	/// Runs the maptos rest api service until crash or shutdown.
+	async fn run_maptos_rest(&self) -> Result<(), anyhow::Error> {
+		self.maptos_rest.run_service().await?;
+		Ok(())
+	}
 }
 
 impl SuzukaPartialNode<SuzukaExecutorV1> {
 	pub async fn try_from_env(
 	) -> Result<(Self, impl Future<Output = Result<(), anyhow::Error>> + Send), anyhow::Error> {
 		let (tx, _) = async_channel::unbounded();
 		let light_node_client = LightNodeServiceClient::connect("http://0.0.0.0:30730").await?;
+
 		let executor = SuzukaExecutorV1::try_from_env(tx)
 			.await
 			.context("Failed to get executor from environment")?;
-		// TODO: switch to real settlement client
-		let settlement_client = MockMcrSettlementClient::new();
-		Self::bound(executor, light_node_client, settlement_client)
+
+		let settlement_client = build_settlement_client().await?;
+		let maptos_rest = MaptosRest::try_from_env(Some(executor.executor.context.clone()))?;
+		Self::bound(executor, light_node_client, settlement_client, maptos_rest)
 	}
 }
+
+pub const ETH_RPC_URL_ENV_VAR: &'static str = "ETH_RPC_URL";
+pub const ETH_WS_URL_ENV_VAR: &'static str = "ETH_WS_URL";
+pub const ETH_MCR_CONTRACT_ADDRESS_VAR: &'static str = "ETH_MCR_CONTRACT_ADDRESS";
+pub const ETH_DEFAULT_TX_GAS_LIMIT_VAR: &'static str = "ETH_DEFAULT_TX_GAS_LIMIT";
+pub const ETH_MAX_TX_SEND_RETRY_VAR: &'static str = "ETH_MAX_TX_SEND_RETRY";
+pub const ETH_VALIDATOR_PRIVATE_ADDRESS_VAR: &'static str = "ETH_VALIDATOR_PRIVATE_ADDRESS";
+
+#[cfg(not(test))]
+use mcr_settlement_client::eth_client::{
+	McrEthSettlementClient, McrEthSettlementConfig, DEFAULT_TX_GAS_LIMIT, MAX_TX_SEND_RETRY,
+	MCR_CONTRACT_ADDRESS,
+};
+#[cfg(not(test))]
+async fn build_settlement_client() -> anyhow::Result<impl McrSettlementClientOperations> {
+	let rpc_url = std::env::var(ETH_RPC_URL_ENV_VAR).unwrap_or("http://localhost:8545".to_string());
+	let ws_url = std::env::var(ETH_WS_URL_ENV_VAR).unwrap_or("ws://localhost:8545".to_string());
+	let gas_limit = std::env::var(ETH_DEFAULT_TX_GAS_LIMIT_VAR)
+		.unwrap_or(DEFAULT_TX_GAS_LIMIT.to_string())
+		.parse()?;
+	let tx_send_nb_retry = std::env::var(ETH_MAX_TX_SEND_RETRY_VAR)
+		.unwrap_or(MAX_TX_SEND_RETRY.to_string())
+		.parse()?;
+	let mrc_contract_address =
+		std::env::var(ETH_MCR_CONTRACT_ADDRESS_VAR).unwrap_or(MCR_CONTRACT_ADDRESS.to_string());
```

Comment:
We have `McrEthSettlementConfig::try_from_env` reading from a differenlty defined set of environment variables, should we avoid duplication here?

This also invites a question as to why the URL and the private address parameters need to be outside `McrEthSettlementConfig` (another name that does not need to be this long in Rust), but I can see a case for exposing more polymorphism on the mcr-settlement-client level.